app/src/main/java/com/example/myweatherapp/MainActivity.kt

```diff
@@ -129,17 +127,14 @@ class MainActivity : AppCompatActivity(), OnItemClickListener, OnSourceClickList
 
         binding.editText.editText.addTextChangedListener(object : TextWatcher {
```

Comment:
that TextWatcher does nothing, do you need it?

Reply:
in progress, left it for some time, now in progress